contracts/BullaFinance.sol

```diff
@@ -116,14 +171,56 @@ abstract contract BullaFinance {
     ///     Allows a debtor to accept a creditor's financing offer and begin payment
     ///     This function will:
     ///         RES1. load the previous claim details and create a new bulla claim specifying `claimAmount` as `originatingClaimAmount + (originatingClaimAmount * terms.interestBPS / 10_000)` and `dueBy` as `term.termLength + block.timestamp`
-    ///         RES2. deletes the `financeTerms` TODO: does this have any potential negative side-effects / drawbacks?
-    ///         RES3. increments `withdrawableFee` by `fee`
+    ///         RES2. deletes the `financeTerms`
+    ///         RES3. increments `unharvestedTransactions`
     ///         RES4. pays `downPayment` amount on the newly created claim
     ///         RES5. emits a LoanAccepted event with the `originatingClaimId` and the new claimId as `financedClaimId`
+    ///         RETURNS: the newly created claimId
     ///     Given the following:
     ///         P1. msg.sender has approved BullaFinance to spend at least `downPayment` amount of the underlying claim's denominating ERC20 token
     ///         P2. `financingTerms[claimId].termLength != 0` (offer exists)
-    ///         P3. `msg.sender == claim.debtor`
-    ///         P4. `downPayment >= (claimAmount * minDownPaymentBPS / 10_000)` && `downPayment < claimAmount + (claimAmount * minDownPaymentBPS / 10_000) (not overpaying or underpaying)
-    function acceptLoan(uint256 claimId, uint256 downPayment) public virtual;
+    ///         P3. `claim.status == ClaimStatus.Pending`
+    ///         P4. `msg.sender == claim.debtor`
+    ///         P5. `downPayment >= (claimAmount * minDownPaymentBPS / 10_000)` && `downPayment < claimAmount + (claimAmount * interestBPS / 10_000) (not overpaying or underpaying)
+    function acceptFinancing(
+        uint256 claimId,
+        uint256 downPayment,
+        string memory newDescription
+    ) public returns (uint256) {
+        Claim memory claim = bullaClaim.getClaim(claimId);
+        FinanceTerms memory terms = financeTermsByClaimId[claimId];
+
+        if (claim.status != Status.Pending) revert CLAIM_NOT_PENDING();
+        if (claim.debtor != msg.sender) revert NOT_DEBTOR();
+        if (terms.termLength == 0) revert NO_FINANCE_OFFER();
+        if (downPayment < ((claim.claimAmount * terms.minDownPaymentBPS) / MAX_BPS)) revert UNDER_PAYING();
+        if (downPayment > claim.claimAmount + ((claim.claimAmount * terms.interestBPS) / MAX_BPS)) revert OVER_PAYING();
+
+        address creditor = IERC721(address(bullaClaim)).ownerOf(claimId);
+        string memory tokenURI = ERC721URIStorage(address(bullaClaim)).tokenURI(claimId);
+
+        uint256 financedClaimId = bullaClaim.createClaimWithURI({
+            creditor: creditor,
+            debtor: claim.debtor,
+            description: newDescription,
+            claimAmount: claim.claimAmount + ((claim.claimAmount * terms.interestBPS) / MAX_BPS), // add interest to the new claim
```

Comment:
you need to apply interest on the loan value (claim amount - down payment)

Reply:
Oh right - since you're only borrowing off the amount - down